turbinia/config/__init__.py

```diff
@@ -191,6 +194,85 @@ def ValidateAndSetConfig(_config):
       setattr(sys.modules[__name__], var, getattr(_config, var))
 
 
+class TurbiniaRecipe(object):
+  """ Base class for Turbinia recipes
+
+  Attributes
+      recipe_file (str): name of the recipe file to be loaded.
+      jobs_allowlist (list): A whitelist for Jobs that will be allowed to run.
+      jobs_denylist (list): A blacklist for Jobs that will not be
+      allowed to run.
+      filter_patterns_file (str): Path to a file containing newline separated
+      string patterns with which to filter text based evidence.
+      task_recipes (dict): Object containing a task specific recipe for
+      each of the tasks invoked in the Turbinia recipe.
+"""
+
+  def __init__(self, recipe_file):
+    self.recipe_file = recipe_file
+    self.name = ""
+    self.jobs_allowlist = []
+    self.jobs_denylist = []
+    self.filter_patterns = []
```

Comment:
I think this is actually only used by the `grep` task right now, so we could probably convert this to a regular recipe item for that task rather than being a top level variable here.

Reply:
References to patterns file eliminated from turbiniactl and recipe parser. 

Made some preliminary modifications to the grep task to allow it to work with the task_config as well as recipes, moved the logic to read a patterns file to this task. 

Thinking about how this would work with docker, should we give "TextFile" and "PlasoCsvFile" source_path attributes to make the file itself accesible to the task?

Reply:
This does not make sense re-reading. I don't think the grep task would be normally run using a docker image :).

Have put in a sample recipe for grep. The general need for giving evidence types that will be consumed by docker a source_path still makes sense though. Not relevant to grep.

Reply:
If we keep filter patterns as a globals recipe item, we can read the patterns into a list and pass it along with the evidence recipe, sparing the need to ensure the file is available to the worker. Keeping it as a globals recipe item pending your input.